ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-history/metacard-history.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+import fetch from '../../utils/fetch'
+const store = require('../../../js/store.js')
+const Common = require('../../../js/Common.js')
+const ResultUtils = require('../../../js/ResultUtils.js')
+const moment = require('moment')
+const announcement = require('component/announcement')
+import MetacardHistoryPresentation from '../../presentation/metacard-history'
+
+type Props = {
+  selectionInterface: any
+}
+
+type State = {
+  model: Backbone.Model
+  history: any
+  selectedVersion: any
+  loading: boolean
+}
+
+class MetacardHistory extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    let model
+    if (props.selectionInterface) {
+      model = props.selectionInterface.getSelectedResults().first()
+    } else {
+      model = store.getSelectedResults().first()
+    }
+
+    this.state = {
+      model: model,
+      history: [],
+      selectedVersion: undefined,
+      loading: true,
+    }
+
+    this.loadData()
+  }
+
+  loadData() {
```

Comment:
✏️ This type of logic is best done in a `componentDidMount` lifecycle event

Reply:
done